apps/oxlint/src/lint/mod.rs

```diff
@@ -184,6 +190,44 @@ impl LintRunner {
         }
         diagnostic_service
     }
+
+    // moved into a separate function for readability, but it's only ever used
+    // in one place.
+    #[inline]
```

Comment:
Probably don't need to add these kind of optimization. Trust the compiler (LLVM) ;-)